packages/block-editor/src/hooks/duotone.js

```diff
@@ -245,6 +249,49 @@ function scopeSelector( scope, selector ) {
 	return selectorsScoped.join( ', ' );
 }
 
+function BlockDuotoneStyles( { name, duotoneStyle, id } ) {
+	const duotonePalette = useMultiOriginPresets( {
+		presetSetting: 'color.duotone',
+		defaultSetting: 'color.defaultDuotone',
+	} );
+
+	const element = useContext( BlockList.__unstableElementContext );
+
+	// Portals cannot exist without a container.
+	// Guard against empty Duotone styles.
+	if ( ! element || ! duotoneStyle ) {
+		return null;
+	}
+
+	let colors = duotoneStyle;
+
+	if ( ! Array.isArray( colors ) && colors !== 'unset' ) {
+		colors = getColorsFromDuotonePreset( colors, duotonePalette );
+	}
+
+	const duotoneSupportSelectors = getBlockSupport(
+		name,
+		'color.__experimentalDuotone'
```

Comment:
I found this line confusing because it seems related to block _supports_ but in actual fact it's just retrieving the CSS selectors defined as targets for the duotone. It will be better once we move this support to a dedicated `filters.duotone` key.